src/ProjectModule/Components/ProjectForm/ProjectForm.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react'
+import styles from './ProjectForm.module.css'
+import { useNavigate, useParams } from 'react-router-dom'
+import { useForm } from 'react-hook-form';
+import axios from 'axios';
+import { baseUrl } from '../../../Constants/Components/Urls';
+import { toast } from 'react-toastify';
+import Loader from "../../../SharedMoudule/Components/Loading/Loading";
+
+
+export default function ProjectForm() {
+    const navigate=useNavigate();
+    const {register,handleSubmit,formState:{errors}}=useForm();
+    const token=localStorage.getItem('userToken');
+    const {action}=useParams();
+    const [currentProject,setCurrentProject]=useState(null);
+    const curruntProjectId=localStorage.getItem('curruntProjectId');
+    const [isLoading, setIsLoading] = useState(false);
+
+    const onSubmitAdd= async(data:any)=>{
+        setIsLoading(true);
+        // console.log('Add');
+        // console.log(data);
+        try {
+            let response=await axios.post(`${baseUrl}/Project`,data,{headers:{Authorization:token}});
+            console.log('add project response ',response);
+            toast.success('Project Added Successfuly');
+            navigate('/dashboard/project-list');
+        } catch (error) {
+            setIsLoading(false);
+            console.log(error);
+            toast.error(error?.response?.data?.message||'Somthing went wrong!');
+        }
+        
+    }
+
+    const onSubmitUpdate= async(data:any)=>{
+        setIsLoading(true);
+        // console.log('Update data');
+        // console.log(data);
+        try {
+            let response=await axios.put(`${baseUrl}/Project/${curruntProjectId}`,{'title':data.title?data.title:currentProject.title,'description':data.description?data.description:currentProject.description},{headers:{Authorization:token}});
```

Comment:
It seems that you are not using a code formatter,  consider **using** prettier for better code formatting and consistency between the team. It will also make the code easier to read.

You can look at this [link](https://youtu.be/DqfQ4DPnRqI?si=JldZPLCgvVEuhr6U) for more about that.

Reply:
why not write the data object instead of accessing its individual attributes?
The data object returning from react-hook-form will contain the input fields that we entered with the registered names.

Reply:
because in update situation if the user didn't change any input the data object will return as an empty object so i did that to fix this issue.

if you have another solution for this please tell me to handle it.

Reply:
@SalmaAboArab 
This should fix the problem: [Link](https://github.com/SalmaAboArab/Project-Management-System/pull/5/files/e34ccad889ac98eb709938d3cef58241d0f15dfc#r1536831244)